### PR TITLE
Add Logging to docs

### LINKED ISSUE #67: Adding Log section to documentation
Will add this the next days

### PR DESCRIPTION
**Pull request by mimugmail**
Closes #67

### CODE REVIEW

**Comment by mimugmail** [comment]
@MichaelDeciso I'd be happy for a review when you find the time <3

File: source/manual/logging.rst [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+   /var/log/squid/cache.log (text)
+   /var/log/squid/store.log (text)
+
+------------
+Circular Log
+------------
+
+Most of the core features log to circular log files so they will not grow bigger
+then a predefined size. You can tune this value via **System->Settings->Logging**.
+There you can also disable the writing of logs to disk or reset them all.
```
Comment (fichtner) [inline source/manual/logging.rst]: Reading a log from the command line:

    # clog /path/to/log

Interactive tailing:

    # clog -f /path/to/log

Comment (mimugmail) [reply source/manual/logging.rst]: fixed

File: source/manual/logging.rst [hunk 2]
```diff
@@ -0,0 +1,135 @@
+==============
+System Logging
+==============
+
+When troubleshooting problems with your Firewall you are heavily constrained to
```
Comment (MichaelDeciso) [inline source/manual/logging.rst]:
```suggestion
When troubleshooting problems with your firewall, it's very likely you have to check
```

File: source/manual/logging.rst [hunk 3]
```diff
@@ -0,0 +1,135 @@
+==============
+System Logging
+==============
+
+When troubleshooting problems with your Firewall you are heavily constrained to 
+the logs available on your system. Via the UI you can find them distributed within
```
Comment (MichaelDeciso) [inline source/manual/logging.rst]:
```suggestion
the logs available on your system. In the UI of OPNsense, the log files are generally grouped
```

File: source/manual/logging.rst [hunk 4]
```diff
@@ -0,0 +1,135 @@
+==============
+System Logging
+==============
+
+When troubleshooting problems with your Firewall you are heavily constrained to 
+the logs available on your system. Via the UI you can find them distributed within 
+the depending Menues which are:
```
Comment (MichaelDeciso) [inline source/manual/logging.rst]:
```suggestion
with the settings of the component they belong to. The log files can be found here:
```

File: source/manual/logging.rst [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+
+------
+System 
+------
+
+============================= ================================ =============================================================
+ **System Log**                **System->Log Files->General**   *Most of all system related events go here*
+ **Backend / config daemon**   **System->Log Files->Backend**   *Here you can find logs for config generation of API usage*
+ **Web GUI**                   **System->Log Files->Web GUI**   *Lighttpd, the webserver of OPNsense itslef logs here*
```
Comment (MichaelDeciso) [inline source/manual/logging.rst]:
```suggestion
 **Web GUI**                   **System->Log Files->Web GUI**   *Lighttpd, the webserver of OPNsense itself, logs here*
```

File: source/manual/logging.rst [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+   /var/log/gateways.log (clog)
+   /var/log/routing.log (clog)
+
+----------
+Interfaces 
+----------
+
+==================== ========================================== ===================================================================
+ **Wireless**         **Interfaces->Wireless->Log File**         *When using wireless features of OPNsense you find the logs here*
+ **Point-to-Point**   **Interfaces->Point-to-Point->Log File**   *PPP dialup logs like PPPoE get's here*
```
Comment (MichaelDeciso) [inline source/manual/logging.rst]:
```suggestion
 **Point-to-Point**   **Interfaces->Point-to-Point->Log File**   *PPP dialup logs like PPPoE are found here*
```

File: source/manual/logging.rst [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+   Log files on file system:
+   /var/log/wireless.log (clog)
+   /var/log/ppps.log (clog)
+
+--------
+Firewall 
+--------
+
+================ ===================================== =============================================================================
+ **Live View**    **Firewall->Log Files->Live View**    *Dynamically updateing windows of logs services search filter and coloring*
```
Comment (MichaelDeciso) [inline source/manual/logging.rst]: Not 100% sure what you're trying to say here. Could you explain it, so I can give a good recommendation?

File: source/manual/logging.rst [hunk 8]
```diff
@@ -0,0 +1,135 @@
...
+   Log files on file system:
+   /var/log/ipsec.log (clog)
+   /var/log/openvpn.log (clog)
+
+--------
+Services
+--------
+
+========================= ============================================= ======================================================
+ **Captive Portal**        **Services->Captive Portal->Log File**        *Events from Captive Portal goes here*
```
Comment (MichaelDeciso) [inline source/manual/logging.rst]:
```suggestion
 **Captive Portal**        **Services->Captive Portal->Log File**        *Events from Captive Portal go here*
```

File: source/manual/logging.rst [hunk 9]
```diff
@@ -0,0 +1,135 @@
...
+ **DHCPv4**                **Services->DHCPv4->Log File**                *DHCP events get logged here*
+ **Dnsmasq DNS**           **Services->Dnsmasq DNS->Log File**           *The DNSmasq Forwarder logs*
+ **HAProxy**               **Services->HAProxy->Log File**               *The logs of the Reverse Proxy*
+ **Intrusion Detection**   **Services->Intrusion Detection->Log File**   *Suricata Logs are here*
+ **Network Time**          **Services->Network Time->Log File**          *NTP daemon logs*
+ **Unbound DNS**           **Services->Unbound DNS->Log File**           *Unbound resolver logs can be found here*
+ **Web Proxy**             **Services->Web Proxy->Log File**             *Squid logs can be found here, three tabs available*
```
Comment (MichaelDeciso) [inline source/manual/logging.rst]: It's better to explain what these three tabs contain.

File: source/manual/logging.rst [hunk 10]
```diff
@@ -0,0 +1,135 @@
...
+   /var/log/squid/access.log (text)
+   /var/log/squid/cache.log (text)
+   /var/log/squid/store.log (text)
+
+------------
+Circular Log
+------------
+
+Most of the core features log to circular log files so they will not grow bigger
+then a predefined size. You can tune this value via **System->Settings->Logging**.
```
Comment (MichaelDeciso) [inline source/manual/logging.rst]:
```suggestion
than a predefined size. You can tune this value via **System->Settings->Logging**.
```

File: source/manual/logging.rst [hunk 11]
```diff
@@ -0,0 +1,135 @@
...
+There you can also disable the writing of logs to disk or reset them all.
```
Comment (MichaelDeciso) [inline source/manual/logging.rst]:
```suggestion
There, you can also disable the writing of logs to disk or reset them all.
```

File: source/manual/logging.rst [hunk 12]
```diff
@@ -0,0 +1,135 @@
...
+
+.. code-block:: sh
+
+    clog -f /path/to/log
+
+------
+Syslog
+------
+
+In **System->Settings->Logging** you can also set a remote log server where to send
```
Comment (MichaelDeciso) [inline source/manual/logging.rst]:
```suggestion
In **System->Settings->Logging** you can also set a remote log server to send
```

File: source/manual/logging.rst [hunk 13]
```diff
@@ -0,0 +1,135 @@
...
+your log files. Choose a **Source Address** if needed and set a **Remote Syslog Server**.
```
Comment (MichaelDeciso) [inline source/manual/logging.rst]:
```suggestion
your log files to. Choose a **Source Address** if needed and fill in the **Remote Syslog Server** field.
```

File: source/manual/logging.rst [hunk 14]
```diff
@@ -0,0 +1,135 @@
...
+Via **Remote Syslog Contents** you can choose which kinds of logs to send.
```
Comment (MichaelDeciso) [inline source/manual/logging.rst]:
```suggestion
Via **Remote Syslog Contents** you can choose which kind of logs to send.
```

File: source/manual/logging.rst [hunk 15]
```diff
@@ -0,0 +1,135 @@
...
+
+-----------
+Plugin Logs
+-----------
+
+Many plugins have their own logs, via GUI you can find them it their own menu.
```
Comment (MichaelDeciso) [inline source/manual/logging.rst]:
```suggestion
Many plugins have their own logs. In the UI, they are grouped with the settings of that plugin.
```

**Comment by fichtner** [comment]
ok, ready for merge?

**Comment by MichaelDeciso** [comment]
I was born ready. I'm Ron ****ing Swanson.

(Not sure if that was directed at me or at the other Michael, but just to be sure.)

**Comment by fichtner** [comment]
I was addressing Michael. Haha, sorry, I couldn't resist. :)

File: source/manual/logging.rst [hunk 16]
```diff
@@ -0,0 +1,135 @@
...
+
+.. Note::
+   Log files on file system:
+   /var/log/portalauth.log (clog)
+   /var/log/dhcpd.log (clog)
+   /var/log/dnsmasq.log (clog)
+   /var/log/haproxy.log (clog)
+   /var/log/ntpd.log (clog)
+   /var/log/suricata.log (text)
+   /var/log/suricata.syskig.log (clog)
```
Comment (fichtner) [inline source/manual/logging.rst]: I think this means suricata.syslog.log, but that file doesn't exist anymore and suricata.log is clog instead?

Comment (mimugmail) [reply source/manual/logging.rst]: Now it's ready .. you're right. I think there is more to add regarding syslog, but this is a good starting point.

File: source/manual/logging.rst [hunk 17]
```diff
@@ -0,0 +1,134 @@
...
+.. Note::
+   Log files on file system:
+   /var/log/portalauth.log (clog)
+   /var/log/dhcpd.log (clog)
+   /var/log/dnsmasq.log (clog)
+   /var/log/haproxy.log (clog)
+   /var/log/ntpd.log (clog)
+   /var/log/suricata.log (clog)
+   /var/log/resolver.log (clog)
+   /var/log/squid.syslog.log (clog)
```
Comment (fichtner) [inline source/manual/logging.rst]: this is also a spurious file, better not to list it

Comment (fichtner) [reply source/manual/logging.rst]: yes, was removed in feb '18 https://github.com/opnsense/core/commit/64ad42d3498f#diff-83987cc3cb034035cd31901cf10701b9L54

Comment (fichtner) [reply source/manual/logging.rst]: neat, did this myself :>

**Comment by fichtner** [comment]
Merged, thanks!

**Pull request merged**